mongoo/mongoo.go

```diff
@@ -0,0 +1,106 @@
+package mongoo
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/faabiosr/cachego"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type (
+	mongoo struct {
+		collection *mongo.Collection
+	}
+
+	mongooContent struct {
+		Duration int64
+		Key      string `bson:"_id"`
+		Value    string
+	}
+)
+
+// New creates an instance of Mongo cache driver
+func New(collection *mongo.Collection) cachego.Cache {
+	return &mongoo{collection}
+}
+
+func (m *mongoo) Contains(key string) bool {
+	_, err := m.Fetch(key)
+	return err == nil
+}
+
+// Delete the cached key from Mongo storage
+func (m *mongoo) Delete(key string) error {
+	_, err := m.collection.DeleteOne(context.TODO(), bson.M{"_id": bson.M{"$eq": key}})
+	return err
+}
+
+// Fetch retrieves the cached value from key of the Mongo storage
+func (m *mongoo) Fetch(key string) (string, error) {
+	content := &mongooContent{}
+
+	if result := m.collection.FindOne(context.TODO(), bson.M{"_id": bson.M{"$eq": key}}); result != nil {
```

Comment:
Yes, good to avoid nested ifs. While I'm developing this, I had the same question. Nested ifs or multiple return statements. I don't have much experience in go lang. What is the best practice in Go? Avoiding nested ifs?   Nested ifs seem to harm the readability too.

I'll change the code to return early.

Reply:
Well, there is not a handbook for it 😄, nested if makes the code less readable, and multiple returns states sounds simpler to understand.